File: 2025/problems/vasaljos/submissions/accepted/atli.cpp
#include<iostream>
#include<string>
using namespace std;

void guess(int a, int b) {
    cout << a << ' ' << b << '\n' << flush;
    string s; cin >> s;
    if(s == "Ljos!") exit(0);
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n; cin >> n;
    if(n == 4) {
        for(int i = 0; i < 4; ++i)
            for(int j = i + 1; j < 4; ++j)
                guess(i + 1, j + 1);
    }
    for(int o = 0; o <= 3; o += 3) {
        guess(1 + o, 2 + o);
        guess(1 + o, 3 + o);
        guess(2 + o, 3 + o);
    }
    for(int i = 7; i < n; i += 2)
        guess(i, i + 1);
}

File: 2025/problems/vasaljos/submissions/accepted/atli.py
#!/usr/bin/python3
n = int(input())


def guess(i, j):
    print(i, j, flush=True)
    if input().strip() == "Ljos!":
        exit(0)


if n == 4:
    for i in range(4):
        for j in range(i + 1, 4):
            guess(i + 1, j + 1)
guess(1, 2)
guess(1, 3)
guess(2, 3)
guess(4, 5)
guess(4, 6)
guess(5, 6)
for i in range(7, n, 2):
    guess(i, i + 1)
